tools/c7n_mailer/c7n_mailer/email_delivery.py

```diff
@@ -274,9 +274,12 @@ def get_aws_username_from_event(self, event):
                 format_struct(event)))
             return None
         if identity['type'] == 'AssumedRole':
-            assume_role_msg = 'No ldap uid is associated with AssumedRole: %s' % identity['arn']
-            self.logger.debug(assume_role_msg)
-            return None
+            self.logger.debug('In some cases there is no ldap uid is associated with AssumedRole: %s' % identity['arn'])
+            self.logger.debug('We will try to assume that identity is in the AssumedRoleSessionName')
+            session_name = identity['arn'].split('/')[-1]
```

Comment:
really its username we're after, and returning after extracting from the session name.

super minor, but also rsplit('/', 1)[-1] is how to ensure the minimal amount of work and temp churn.